cli/admin/src/main/java/io/pravega/cli/admin/controller/ControllerCommand.java

```diff
@@ -107,21 +108,44 @@ protected Context createContext() {
      * @return Response for the REST call.
      */
     String executeRESTCall(Context context, String requestURI) {
-        Invocation.Builder builder;
-        String resourceURL = getCLIControllerConfig().getControllerRestURI() + requestURI;
-        WebTarget webTarget = context.client.target(resourceURL);
-        builder = webTarget.request();
-        Response response = builder.get();
+        Response response = getInvocationBuilder(context, requestURI).get();
         printResponseInfo(response);
         return response.readEntity(String.class);
     }
 
+    /**
+     * Generic method to execute a delete request against the Controller and get the response.
+     *
+     * @param context Controller command context.
+     * @param requestURI URI to execute the request against.
+     * @return Response for the REST call.
+     */
+    String executeDeleteRESTCall(Context context, String requestURI) {
+        Response response = getInvocationBuilder(context, requestURI).delete();
+        printResponseInfo(response);
+        return response.readEntity(String.class);
+    }
+
+    /**
+     * Method to get invocationBuilder to execute rest api.
+     * @param context       Controller command context.
+     * @param requestURI    URI to execute the request against.
+     * @return              InvocationBuilder object.
+     */
+    private Invocation.Builder getInvocationBuilder(final Context context, final String requestURI) {
+        String resourceURL = getCLIControllerConfig().getControllerRestURI() + requestURI;
+        WebTarget webTarget = context.client.target(resourceURL);
+        return webTarget.request();
+    }
+
     @VisibleForTesting
     void printResponseInfo(Response response) {
         if (OK.getStatusCode() == response.getStatus()) {
             output("Successful REST request.");
         } else if (UNAUTHORIZED.getStatusCode() == response.getStatus()) {
             output("Unauthorized REST request. You may need to set the user/password correctly.");
+        } else if (NO_CONTENT.getStatusCode() == response.getStatus()) {
```

Comment:
can we move this to `OR` part of `If` case because the output is same in both cases